### DISCUSSION TITLE
[rules] Access Power1 value from another trigger

### DISCUSSION THREAD

**Discussion by BusinessClaes** [arendst/Tasmota#16727] [Q&A]
Hello,

I want my lights to respond to long press in different ways depending on the state of Power1. If lights are on, long press will dim up/down. If lights are off, long press will dim to value 1.

I manage to get this done by storing in Var1 the %value% of Power1#state trigger. Then on long press trigger I launch an event with attribute %var1%. Then on event#longpress=1 or =0 I run the different actions.

But using a variable to constantly store a copy of the value of Power1 seems complicated/unnecessary to me. The value is already stored in memory, so it seems unnecessary to copy that value time after time. Does anyone know it the value of Power1 can be accessed directly, upon another trigger, not only as %value% of the trigger Power1#state?

I have tried %Power1% and %Power1#value% but they do not work…

**Comment by barbudor** [comment]
> I have tried %Power1% and %Power1#value% but they do not work…

Indeed as this doesn't exist

So far, as far as I can see, you've implemented the nicest solution to your problem 👍

**Reply by sfromis** [reply]
Running a rule "too often" is not seen as an "efficiency problem". If you want something more direct, you can use ESP32 and the Berry language. Possibly also the scripting language available on ESP8266 too, but without real experience there, I'm not sure of anything, other than there is quite some learning curve to get started with that.

**Reply by BusinessClaes** [reply]
Yes I know they don’t exist, I guess I was hoping for some magic :)

Actually the question could be wider, kind of “wouldn’t it be great if we could directly access all the variables stored in memory, instead of being restricted to wait for a trigger in order to read the value?”

I guess it is not as simple as it sounds, otherwise it would have been already implemented. I wonder if this could be matter for a feature request or if it is just not realistic…

**Reply by sfromis** [reply]
Tasmota has been tuned for the very limited capacity of the ESP8266, meaning that going from "here's a feasible way of doing it without too much trouble" to "wouldn't it be great if" is normally not taken, as there is no strong case for it. It is not like on a PC where there is basically no "cost" of having convenience features implemented.

**Reply by BusinessClaes** [reply]
Yes, I understand the ESP8266 has its limitations, I actually consider Tasmota to be a masterpiece and I really enjoy using it. But since I don’t have the skills to provide a solution I guess I was hoping for someone to say “well, that’s not difficult to implement at all, we just don’t consider it to be a basic feature. But it sounds interesting and we might look into including this as a possibility for custom builds in the future”.
I hope it doesn’t sound pretentious, it is just a humble wish and I completely understand that it might not be realistic.

I might try with the scripting language as you suggest, but as you say it is a steep learning curve, so I don’t know if I’ll succeed :)
